init_process.go

```diff
@@ -0,0 +1,406 @@
+	arnBucketFile := arn[13:]
+	bucketAndFile := strings.SplitN(arnBucketFile, "/", 2)
+	if len(bucketAndFile) != 2 {
+		log.Fatalf("Unrecognizable arn: %s\n", arn)
+	}
+
+	bucketName := bucketAndFile[0]
+	s3FilePath := bucketAndFile[1]
+
+	// get bucket region
+	input := &s3.GetBucketLocationInput{
+		Bucket: aws.String(bucketName),
+	}
+
+	output, err := s3Client.GetBucketLocation(input)
+	if err != nil {
+		log.Println(bucketName + ":" + s3FilePath)
+		log.Println(err)
+		log.Println("Cannot get bucket region")
+	}
+
+	bucketRegion := aws.StringValue(output.LocationConstraint)
+	// Buckets in Region us-east-1 have a LocationConstraint of null.
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#GetBucketLocationOutput
+	if bucketRegion == "" {
+		bucketRegion = "us-east-1"
+	}
+
+	// create downloader
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(bucketRegion)},
+	)
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot creat a new session")
+	}
+
+	// need to specify session region!
+	s3Downloader := s3manager.NewDownloader(sess)
+
+	// download file and store
+	s3FileName := strings.SplitN(s3FilePath, "/", -1)
+	fileFromS3, err := os.Create(ConfigFileFolderPath + "/" + s3FileName[len(s3FileName)-1])
+	if err != nil {
+		log.Println(err)
+		log.Fatalln("Cannot creat s3 config file to store config info")
+	}
+	defer fileFromS3.Close()
+
+	_, err = s3Downloader.Download(fileFromS3,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucketName),
+			Key:    aws.String(s3FilePath),
+		})
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Cannot download %s from s3\n", s3FileName)
+	}
+}
+
+// create a S3 client as the global S3 client for reuse
+func createS3Client() {
+	region := "us-east-1"
+	if metadataReigon != "" {
+		region = metadataReigon
+	}
+	s3Client = s3.New(session.Must(session.NewSession(&aws.Config{
+		// if not specify region here, missingregion error will raise when get bucket location
+		Region: aws.String(region),
+	})))
+
+	exists3Client = true
+}
+
+// process S3 config files user specified.
+func processS3ConfigFiles(folderPath string) {
+	fileInfos, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		log.Println(err)
+		log.Fatalf("Unable to read config files in %s folder\n", folderPath)
+	}
+
+	for _, file := range fileInfos {
+		content_b, err := ioutil.ReadFile(folderPath + "/" + file.Name())
+		if err != nil {
+			log.Println(err)
+			log.Fatalf("Cannot open file: %s\n", folderPath+"/"+file.Name())
+		}
+		content := string(content_b)
+		if strings.Contains(content, "[PARSER]") {
+			// this is a parser config file. change command
+			updateCommand("/" + folderPath + "/" + file.Name())
+		} else {
+			// this is not a parser config file. @INCLUDE
+			writeInclude("/"+folderPath+"/"+file.Name(), MainConfigFilePath)
+		}
+	}
+}
+
+// add @INCLUDE in main config file to include all customer specified config files
+func writeInclude(configFilePath, mainConfigFilePath string) {
```

Comment:
Why do we need `includeOriginalMainConfigFile` along? both do include thing

Reply:
Make sense.. already modified